collect_app/src/main/java/org/odk/collect/android/widgets/GeoPointWidget.java

```diff
@@ -157,8 +157,12 @@ private void updateButtonLabelsAndVisibility(boolean dataAvailable) {
                 getLocationButton.setVisibility(View.GONE);
             } else {
                 getLocationButton.setVisibility(View.VISIBLE);
-                getLocationButton.setText(getContext().getString(
-                        dataAvailable ? R.string.get_point : R.string.get_point));
```

Comment:
I believe you can get the desired behavior just by changing the first `R.string.get_point` to your new `R.string.change_location`! It looks like `dataAvailable` is equivalent to your test to see if there is an answer already. If you're not familiar with this syntax, it's equivalent to a conditional: https://www.sitepoint.com/java-ternary-operator/

Reply:
@SIMRAN1 Does the above make sense? Please ask if you have any questions.